### PR TITLE
feat(react-query-v4): react-query-next-experimental

### PR DESCRIPTION
**Pull request by 2-NOW**
# Overview

#1155 

Adds a streaming without prefetching feature that brings the [react-query-next-experimental](https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr#experimental-streaming-without-prefetching-in-nextjs) feature available in react-query v5 to v4

- [x] Add a default implementation
  - HydrationStreamProvider
  - ReactQueryStreamedHydration
- [x] ~~Test working well~~ continue with the follow-up.

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

**Comment by 2-NOW** [comment]
Is there a template or rule for creating new package? If there are conventions for the repository here, I'd be happy to know about them

**Comment by manudeli** [comment]
@2-NOW Welcome🙇 I added commit [65d2fc4](https://github.com/toss/suspensive/pull/1161/commits/65d2fc415286051d951136321f38bf7f0aaeff26) to support you to get playground and convention

**Comment by SEOKKAMONI** [comment]
@manudeli It seems that something went wrong due to a forced push. It needs to be checked.

**Comment by 2-NOW** [comment]
>@manudeli It seems that something went wrong due to a forced push. It needs to be checked.

It looks like you changed the history accordingly :) Thanks for checking @SEOKKAMONI

**Comment by manudeli** [comment]
coauthors test

**Comment by manudeli** [comment]
1. Merge origin/main please. we use pnpm@9.7.1 fixed by @gwansikk from now
2. Share your proximate plan on this pull request please. I want to check what you have issue to do this after that date

**Comment by 2-NOW** [comment]
>Share your proximate plan on this pull request please. I want to check what you have issue to do this after that date

I'll be doing some additional work this Sunday to make sure that the promise dehydratedState supported by v5 doesn't break v4's behavior. Once that's confirmed, the POC should be largely resolved.
It's still an unfinished feature with no guaranteed implementation, but would it be okay to merge it into main ? @manudeli

**Comment by manudeli** [comment]
> > Share your proximate plan on this pull request please. I want to check what you have issue to do this after that date
> 
> would it be okay to merge it into main ? @manudeli

Sure, let's merge. to merge this branch, could you resolve conflict first please?

**Comment by 2-NOW** [comment]
> Sure, let's merge. to merge this branch, could you resolve conflict first please?

I've resolved the conflict, please confirm that it's okay to merge. @manudeli

**Comment by manudeli** [comment]
Resolve knip error please

**Comment by 2-NOW** [comment]
> Resolve knip error please

Fixed :D

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "changeset": "changeset",
     "changeset:publish": "pnpm prepack && changeset publish",
     "changeset:version": "changeset version && pnpm i --lockfile-only",
-    "ci:all": "pnpm ci:attw && pnpm ci:eslint && pnpm ci:publint && pnpm ci:sherif && pnpm ci:type && pnpm ci:test && pnpm ci:knip && pnpm ci:bench && pnpm build",
+    "ci:all": "pnpm ci:attw && pnpm ci:eslint && pnpm ci:publint && pnpm ci:sherif && pnpm ci:type && pnpm ci:test && pnpm ci:bench && pnpm build && pnpm ci:knip ",
```
Comment (manudeli) [inline package.json:24]:
```suggestion
    "ci:all": "pnpm ci:attw && pnpm ci:eslint && pnpm ci:publint && pnpm ci:sherif && pnpm ci:type && pnpm ci:test && pnpm ci:knip && pnpm ci:bench && pnpm build",
```

File: packages/react-query-next-experimental/package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  "dependencies": {
+    "@suspensive/react-query-next-experimental-4": "workspace:*",
+    "@suspensive/utils": "workspace:*"
+  },
+  "devDependencies": {
+    "@suspensive/eslint-config": "workspace:*",
+    "@suspensive/tsconfig": "workspace:*",
+    "@suspensive/tsup": "workspace:*",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
```
Comment (manudeli) [inline packages/react-query-next-experimental/package.json:59]:
```suggestion
```

File: packages/react-query-next-experimental-4/package.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  "devDependencies": {
+    "@suspensive/eslint-config": "workspace:*",
+    "@suspensive/tsconfig": "workspace:*",
+    "@suspensive/tsup": "workspace:*",
+    "@tanstack/react-query": "^4.36.1",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
+    "next": "^14.2.5",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1"
```
Comment (manudeli) [inline packages/react-query-next-experimental-4/package.json:64]:
```suggestion
```

File: packages/react-query-next-experimental-4/package.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  "dependencies": {
+    "@suspensive/utils": "workspace:*"
+  },
+  "devDependencies": {
+    "@suspensive/eslint-config": "workspace:*",
+    "@suspensive/tsconfig": "workspace:*",
+    "@suspensive/tsup": "workspace:*",
+    "@tanstack/react-query": "^4.36.1",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
```
Comment (manudeli) [inline packages/react-query-next-experimental-4/package.json:61]:
```suggestion
```

**Review by manudeli — APPROVED** [review]
Let's improve after merging

**Pull request merged**